test/inductor/test_cudagraph_trees.py

```diff
@@ -336,6 +337,47 @@ def foo(mod, x):
 
             self.assertIsNotNone(self.get_manager())
 
+        @parametrize("backend", ("inductor", "cudagraphs"))
+        @torch._dynamo.config.patch("cudagraph_backend_keep_input_mutation", True)
+        def test_mutation_cudagraph_managed_tensors(self, backend):
+            def foo(x):
+                return x + 1
+
+            def mut(x):
+                x.add_(2)
+                return x
+
+            def non_mut(x):
+                return x.add(2)
+
+            mut = get_compile_fn(backend)(mut)
+            foo = get_compile_fn(backend)(foo)
+
+            for i in range(3):
+                torch.compiler.cudagraph_mark_step_begin()
+                inp = torch.rand([4], device="cuda")
+
+                tmp = foo(inp)
+                with capture_stderr() as captured_output:
+                    mut_out = mut(tmp)
+                if i == 0:
+                    FileCheck().check(
+                        "skipping cudagraphs due to mutation on input."
+                    ).check("x.add_(2)").run(captured_output[0])
+                else:
+                    self.assertEqual(mut_out, non_mut(foo(inp)))
+
+            torch.compiler.cudagraph_mark_step_begin()
+            inp = torch.rand([4], device="cuda")
+            tmp = foo(inp)
+            # in this case, what previously a mutated cudagraph managed tensor is no longer,
+            # now its an input from eager we should fallback to inductor without cudagraphs
+            with capture_stderr() as captured_output:
+                mut(tmp.clone())
```

Comment:
Let's save `tmp.clone()` as an intermediary, and check that it is equal to `non_mut(foo(inp)))`